src/validNumberDfa.cpp

```diff
@@ -1,7 +1,7 @@
 #include "states.h"
 #include "validNumberDfa.h"
 
-int error(char c) {
+int validNumberError(char c) {
```

Comment:
The last time we had error inside of the class, it conflicts with the definition of TransitionFunction and I couldn't compile it, unless I use the -fpermissive flag